### PR TITLE
Accept text/html, text/* and */* as Accept header

### LINKED ISSUE #1182: win 10 home ,cant't start
Forbidden

Invalid Accept header to load wrapper

Please report it if you think this an error.

Details:
{
    "GATEWAY_INTERFACE": "CGI/1.1", 
    "HTTP_ACCEPT": "*/*", 
    "HTTP_ACCEPT_ENCODING": "gzip, deflate", 
    "HTTP_ACCEPT_LANGUAGE": "zh-Hans-CN,zh-Hans;q=0.8,en-US;q=0.5,en;q=0.3", 
    "HTTP_CONNECTION": "Keep-Alive", 
    "HTTP_HOST": "127.0.0.1:43110", 
    "HTTP_USER_AGENT": "Mozilla/5.0 (Windows NT 10.0; WOW64; Trident/7.0; Touch; rv:11.0; 2345Explorer/8.8.3.16721) like Gecko", 
    "PATH_INFO": "/1HeLLo4uzjaLetFx6NH3PMwFP3qbRbTf3D", 
    "QUERY_STRING": "", 
    "REMOTE_ADDR": "127.0.0.1", 
    "REMOTE_PORT": "51975", 
    "REQUEST_METHOD": "GET", 
    "SCRIPT_NAME": "", 
    "SERVER_NAME": "JK-GPD-win", 
    "SERVER_PORT": "43110", 
    "SERVER_PROTOCOL": "HTTP/1.1", 
    "SERVER_SOFTWARE": "gevent/1.2 Python/2.7", 
    "arguments": {
        "action": "main", 
        "autodownload_bigfile_size_limit": 1, 
        "batch": false, 
        "bind": null, 
        "bit_resolver": "1Name2NXVi1RDPDgf5617UoW7xA6YrhM9F", 
        "coffeescript_compiler": "type %s | tools\\coffee\\coffee.cmd", 
        "config_file": "C:/ZeroNet/zeronet.conf", 
        "connected_limit": 8, 
        "data_dir": "C:/ZeroNet/data", 
        "db_mode": "speed", 
        "debug": false, 
        "debug_gevent": false, 
        "debug_socket": false, 
        "disable_db": false, 
        "disable_encryption": false, 
        "disable_sslcompression": true, 
        "disable_udp": false, 
        "download_optional": "manual", 
        "end": true, 
        "file_size_limit": 10, 
        "fileserver_ip": "*", 
        "fileserver_port": 15441, 
        "fix_float_decimals": false, 
        "homepage": "1HeLLo4uzjaLetFx6NH3PMwFP3qbRbTf3D", 
        "ip_external": null, 
        "ip_local": [
            "127.0.0.1"
        ], 
        "keep_ssl_cert": false, 
        "language": "zh", 
        "log_dir": "C:/ZeroNet/log", 
        "max_files_opened": 2048, 
        "msgpack_purepython": false, 
        "open_browser": "default_browser", 
        "optional_limit": "10%", 
        "pin_bigfile": 20, 
        "proxy": null, 
        "silent": false, 
        "size_limit": 10, 
        "stack_size": null, 
        "stream_downloads": false, 
        "tor": "enable", 
        "tor_controller": "127.0.0.1:9051", 
        "tor_hs_limit": 10, 
        "tor_proxy": "127.0.0.1:9050", 
        "trackers": [
            "zero://boot3rdez4rzn36x.onion:15441", 
            "zero://zero.booth.moe#f36ca555bee6ba216b14d10f38c16f7769ff064e0e37d887603548cc2e64191d:15441", 
            "udp://tracker.coppersurfer.tk:6969", 
            "udp://tracker.leechers-paradise.org:6969", 
            "udp://9.rarbg.com:2710", 
            "http://tracker.opentrackr.org:1337/announce", 
            "http://explodie.org:6969/announce", 
            "http://tracker1.wasabii.com.tw:6969/announce"
        ], 
        "trackers_file": false, 
        "ui_host": null, 
        "ui_ip": "127.0.0.1", 
        "ui_port": 43110, 
        "ui_restrict": false, 
        "updatesite": "1UPDatEDxnvHDo7TXvq6AEBARfNkyfxsp", 
        "use_openssl": true, 
        "use_tempfiles": false, 
        "verbose": false, 
        "workers": 5
    }, 
    "plugins": [
        "AnnounceZero", 
        "Bigfile", 
        "Cors", 
        "CryptMessage", 
        "FilePack", 
        "MergerSite", 
        "Mute", 
        "Newsfeed", 
        "OptionalManager", 
        "PeerDb", 
        "Sidebar", 
        "Stats", 
        "TranslateSite", 
        "Trayicon", 
        "Zeroname"
    ], 
    "version_gevent": "1.2.1", 
    "version_python": "2.7.13 (v2.7.13:a06454b1afa1, Dec 17 2016, 20:42:59) [MSC v.1500 32 bit (Intel)]", 
    "version_zeronet": "0.6.0 r3128", 
    "wsgi.url_scheme": "http"
}

### PR DESCRIPTION
**Pull request by imachug**
Fixes #1182. Also allows simple curl requests.

### CODE REVIEW

File: src/Ui/UiRequest.py [hunk 1]
```diff
@@ -276,7 +276,7 @@ def actionWrapper(self, path, extra_headers=None):
             if self.isWebSocketRequest():
                 return self.error403("WebSocket request not allowed to load wrapper")  # No websocket
 
-            if "text/html" not in self.env.get("HTTP_ACCEPT", ""):
+            if all([accept not in self.env.get("HTTP_ACCEPT", "") for accept in ("text/html", "text/*", "*/*")]):
```
Comment (skwerlman) [inline src/Ui/UiRequest.py]: maybe rename `accept` to `mime_type` or something similar to make this line clearer in intent

**Comment by HelloZeroNet** [comment]
The content type restriction is present, to avoid site addition with a simple img tag eg.: ``
You can use the /raw/ prefix to load content using curl. eg.: http://127.0.0.1:43110/raw/1anything...

**Comment by imachug** [comment]
@HelloZeroNet ! And what about fixing #1182 then?!

File: src/Ui/UiRequest.py (hunk 1 above)
Comment (imachug) [reply src/Ui/UiRequest.py]: @skwerlman Done. Though @HelloZeroNet refused to merge, so I'm not sure if this is helpful.

**Comment by HelloZeroNet** [comment]
It looks like some weird browser: `"Mozilla/5.0 (Windows NT 10.0; WOW64; Trident/7.0; Touch; rv:11.0; 2345Explorer/8.8.3.16721) like Gecko"`
Not sure what is "2345Explorer/8.8.3.16721", but I just tested in IE11 (based on `Trident/7.0; Touch; rv:11.0;` part) and it works

**Comment by imachug** [comment]
@HelloZeroNet This makes sense. I asked @jarodlee about their OS/browser.

**Comment by skwerlman** [comment]
a brief google says its a PUP

**Comment by imachug** [comment]
@MuxZeroNet said that this happens even on Internet Explorer - [An old post about this](https://blogs.msdn.microsoft.com/ieinternals/2009/07/01/ie-and-the-accept-header/) - so maybe we need another way to detect if this is `img` or a link.

**Comment by celeron533** [comment]
@HelloZeroNet 
>Not sure what is "2345Explorer/8.8.3.16721"

It's a browser owned by `baidu.com`

**Pull request closed without merging**